app/cbt/utils/text_manipulations.py

```diff
@@ -0,0 +1,117 @@
+from config.get_config import config_object
+import aiohttp
+from utils.utils import logger
+
+
+async def get_languages() -> dict:
+    async with aiohttp.ClientSession(
+        base_url=config_object.libretranslate_server_config.url
+    ) as session:
+        languages: list
+        async with session.get("/languages") as response:
+            match response.status:
+                case 200:
+                    logger.debug(f"Got response from {response.real_url}")
+                    try:
+                        languages = await response.json()
+                    except Exception as e:
+                        logger.error(f"Failed to fetch languages with error {e}")
+                        raise e
+                case _:
+                    logger.error(
+                        f"Call to {response.real_url} failed with error code {response.status}"
+                    )
+                    logger.debug(await response.json())
+                    raise ConnectionError(
+                        f"Call to {response.real_url} failed with error code {response.status}"
+                    )
+
+        if len(languages) == 0:
+            logger.error("Failed to find any languages")
+            raise ConnectionError(
+                f"Call to {response.real_url} failed to fetch languages"
+            )
+
+        # build language dict
+        languages_dict = {
+            language["code"]: {"name": language["name"], "targets": language["targets"]}
+            for (language) in languages
+        }
+
+        return languages_dict
+
+
+async def translate_text(
+    text: str, target_language_in: str = None, source_language_in: str = "auto"
+) -> str:
+    if target_language_in is None:
+        return text
+    target_language: str = target_language_in.casefold()
+    source_language: str = source_language_in.casefold()
+    logger.info(f"Invoking translator {source_language}->{target_language}")
+
+    async with aiohttp.ClientSession(
+        base_url=config_object.libretranslate_server_config.url
+    ) as session:
+        real_source_language = "en"
+        if source_language == "auto":
+            async with session.post("/detect", json={"q": text}) as response:
+                match response.status:
+                    case 200:
+                        logger.debug(f"Got response from {response.real_url}")
+                        try:
+                            real_source_language = (await response.json())[0][
+                                "language"
```

Comment:
This seems like a magic string here: `"language"`
We shouldn't be using magic strings, please refrain from doing so.

Reply:
This comes from https://libretranslate.com/docs/
cry